raster-foundry/raster-foundry
# Create error handling ADT for better error handling

## Overview

This PR adds an a `BacksplashError` sealed trait and fancy magic implicit error handling so that we'll always know when we've failed to handle one of the types of errors that we know about.

### Checklist


### Notes

## Testing Instructions

 * comment out one of the matches in the `case` statement that checks what kind of errors we have
 * compile
 * observe that the compiler is mad
 * grab a tile request that succeeds from the frontend
 * make some variations on that request and check the errors you get back:
   * todo -- fill in

Closes #3957
